Examples/Algorithms/Digitization/src/DigitizationOptions.cpp

```diff
@@ -11,6 +11,7 @@
 #include "Acts/Definitions/TrackParametrization.hpp"
 #include "Acts/Definitions/Units.hpp"
 #include "Acts/Utilities/Logger.hpp"
+#include "ActsExamples/Digitization/JsonDigitizationConfig.hpp"
```

Comment:
I see the reasoning here, but I am not sure we should drag the `Json` config into the `Algorithm/Digitizatino` folder.

Reply:
The problem here is that keeping it in the JSON folder created a circular dependency when I tried to use its function in the Digitization package. There's probably other ways to organize the code to avoid this, tho.

Reply:
I moved the json-specific stuff back to the json module. I've added a constructor for the AlgorithmConfig class (to be renamed) that takes a geometrymap<cfg> object as an argument that's moved into the config, to keep things simple.

One consequence of this refactoring is that the SmearingAlgorithm config via .json is completely overridden by the command-line parameters if both are present; I've added a warning for such cases.